src/sdam/topology.ts

```diff
@@ -930,8 +933,13 @@ function processWaitQueue(topology: Topology) {
     let selectedDescriptions;
     try {
       const serverSelector = waitQueueMember.serverSelector;
+      const previousServer = waitQueueMember.previousServer;
       selectedDescriptions = serverSelector
-        ? serverSelector(topology.description, serverDescriptions)
+        ? serverSelector(
+            topology.description,
+            serverDescriptions,
+            previousServer ? [previousServer] : []
```

Comment:
Do we basically set the `deprioritized` value here? Does it mean that we encounter only the most recent previousServer, not all that we tried and havend't succeeded before?

Reply:
Yes currently we only encounter the most recent previous server, as we only retry once right now. Once CSOT is implemented, then we update this to push each previous previous server to the deprioritised list since we then can retry multiple times. I made this an array in anticipation of the CSOT work even though right now it will contain only 1.

Reply:
Got it! Thx for the detailed explanation ✨